### PR TITLE
Added cloneTo method to support trajectory point copying.

### PR DESCRIPTION
**Pull request by shaun-edwards**
Addresses issue #27.  Refactored/expanded test suite.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
@dpsolomon, a quick one to check.

File: descartes_core/test/trajectory_pt.cpp [hunk 1]
```diff
@@ -61,23 +74,31 @@ typedef Types<CartTrajectoryPt, JointTrajectoryPt> Implementations;
...
-}
 
-TYPED_TEST(TrajectoryPtTest, idCopy) {
-  this->lhs_ = this->rhs_;
-  EXPECT_FALSE(this->lhs_->getID().is_nil());
-  EXPECT_FALSE(this->rhs_->getID().is_nil());
+  //Copied objects should always be the same
+  //TODO: The equal comparison fails for some unknown reason.  In the debugger
+  //the values are equal, so this may be a problem with eigen.
+  //EXPECT_EQ(this->lhs_, this->lhs_copy_);
```
Comment (dpsolomon) [inline descartes_core/test/trajectory_pt.cpp]: I think you are comparing pointers, which should return false in this case. Did you try 
`EXPECT_EQ(*(this->lhs_), *(this->lhs_copy_));`?

**Comment by dpsolomon** [comment]
+1 @shaun-edwards . If you want to fix the test first let me know, if you don't care about it too much I'll just merge it.

**Comment by shaun-edwards** [comment]
Pointers...well that's embarrassing.  I fixed the tests to reflect what I really wanted...and then commented them out because that functionality is missing.  I added issue #32 to address this in the future.  @jrgnicho needs the clone functionality ASAP, so I'd prefer to accept the PR as is.  @dpsolomon, can you merge?

**Comment by dpsolomon** [comment]
Next time I'll send a private note ;)

**Pull request merged**